evm/src/arithmetic/columns.rs

```diff
@@ -100,18 +101,28 @@ pub(crate) const MODULAR_INPUT_0: Range<usize> = GENERAL_INPUT_0;
 pub(crate) const MODULAR_INPUT_1: Range<usize> = GENERAL_INPUT_1;
 pub(crate) const MODULAR_MODULUS: Range<usize> = GENERAL_INPUT_2;
 pub(crate) const MODULAR_OUTPUT: Range<usize> = GENERAL_INPUT_3;
-pub(crate) const MODULAR_QUO_INPUT_LO: Range<usize> = AUX_INPUT_0_LO;
+pub(crate) const MODULAR_QUO_INPUT: Range<usize> = GENERAL_INPUT_4_DBL;
+pub(crate) const MODULAR_OUT_AUX_RED: Range<usize> = AUX_INPUT_0;
 // NB: Last value is not used in AUX, it is used in MOD_IS_ZERO
-pub(crate) const MODULAR_QUO_INPUT_HI: Range<usize> = AUX_INPUT_0_HI;
-pub(crate) const MODULAR_AUX_INPUT: Range<usize> = AUX_INPUT_1.start..AUX_INPUT_1.end - 1;
-pub(crate) const MODULAR_MOD_IS_ZERO: usize = AUX_INPUT_1.end - 1;
-pub(crate) const MODULAR_OUT_AUX_RED: Range<usize> = AUX_INPUT_2;
+pub(crate) const MODULAR_MOD_IS_ZERO: usize = AUX_INPUT_1.start;
+pub(crate) const MODULAR_AUX_INPUT_LO: Range<usize> = AUX_INPUT_1.start + 1..AUX_INPUT_1.end;
+pub(crate) const MODULAR_AUX_INPUT_HI: Range<usize> = AUX_INPUT_2;
+// Must be set to MOD_IS_ZERO for DIV operation i.e. MOD_IS_ZERO * lv[IS_DIV]
+pub(crate) const MODULAR_DIV_DENOM_IS_ZERO: usize = AUX_INPUT_2.end;
 
-#[allow(unused)] // TODO: Will be used when hooking into the CPU
 pub(crate) const DIV_NUMERATOR: Range<usize> = MODULAR_INPUT_0;
-#[allow(unused)] // TODO: Will be used when hooking into the CPU
 pub(crate) const DIV_DENOMINATOR: Range<usize> = MODULAR_MODULUS;
 #[allow(unused)] // TODO: Will be used when hooking into the CPU
-pub(crate) const DIV_OUTPUT: Range<usize> = MODULAR_QUO_INPUT_LO;
-
-pub const NUM_ARITH_COLUMNS: usize = START_SHARED_COLS + NUM_SHARED_COLS;
+pub(crate) const DIV_OUTPUT: Range<usize> =
+    MODULAR_QUO_INPUT.start..MODULAR_QUO_INPUT.start + N_LIMBS;
+
+// Need one column for the table, then two columns for every value
+// that needs to be range checked in the trace, namely the permutation
+// of the column and the permutation of the range. The two
+// permutations associated to column i will be in columns RC_COLS[2i]
+// and RC_COLS[2i+1].
+pub(crate) const NUM_RANGE_CHECK_COLS: usize = 1 + 2 * NUM_SHARED_COLS;
+pub(crate) const RANGE_COUNTER: usize = START_SHARED_COLS + NUM_SHARED_COLS;
+pub(crate) const RC_COLS: Range<usize> = RANGE_COUNTER..RANGE_COUNTER + 2 * NUM_SHARED_COLS;
```

Comment:
Shouldn't `RC_COLS` be `RANGE_COUNTER + 1..RANGE_COUNTER + 1 + 2 * NUM_SHARED_COLS`?